File: TicTacToeApp/Server/AP1_Jv_T03.ID_1369281-1/src/Project_03/src/main/java/org/web/model/LeaderboardDto.java
// ./main/java/org/web/model/LeaderboardDto.java
package org.web.model;

public class LeaderboardDto {
    private String username;
    private int gamesPlayed;
    private int gamesWon;
    private int gamesLost;
    private int gamesDrawn;
    private double winRate;
    private int rating; // рейтинг для сортировки

    public LeaderboardDto() {}

    public LeaderboardDto(String username, int gamesPlayed, int gamesWon,
                          int gamesLost, int gamesDrawn, double winRate, int rating) {
        this.username = username;
        this.gamesPlayed = gamesPlayed;
        this.gamesWon = gamesWon;
        this.gamesLost = gamesLost;
        this.gamesDrawn = gamesDrawn;
        this.winRate = winRate;
        this.rating = rating;
    }

    // Геттеры и сеттеры
    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }

    public int getGamesPlayed() { return gamesPlayed; }
    public void setGamesPlayed(int gamesPlayed) { this.gamesPlayed = gamesPlayed; }

    public int getGamesWon() { return gamesWon; }
    public void setGamesWon(int gamesWon) { this.gamesWon = gamesWon; }

    public int getGamesLost() { return gamesLost; }
    public void setGamesLost(int gamesLost) { this.gamesLost = gamesLost; }

    public int getGamesDrawn() { return gamesDrawn; }
    public void setGamesDrawn(int gamesDrawn) { this.gamesDrawn = gamesDrawn; }

    public double getWinRate() { return winRate; }
    public void setWinRate(double winRate) { this.winRate = winRate; }

    public int getRating() { return rating; }
    public void setRating(int rating) { this.rating = rating; }
}

File: TicTacToeApp/app/src/main/java/com/example/tictactoeapp/data/model/LeaderboardDto.kt
// ./main/java/com/example/tictactoeapp/data/model/LeaderboardDto.kt
package com.example.tictactoeapp.data.model

import com.google.gson.annotations.SerializedName

data class LeaderboardDto(
    @SerializedName("username")
    val username: String? = null,

    @SerializedName("gamesPlayed")
    val gamesPlayed: Int = 0,

    @SerializedName("gamesWon")
    val gamesWon: Int = 0,

    @SerializedName("gamesLost")
    val gamesLost: Int = 0,

    @SerializedName("gamesDrawn")
    val gamesDrawn: Int = 0,

    @SerializedName("winRate")
    val winRate: Double = 0.0,

    @SerializedName("rating")
    val rating: Int = 0
)